### PR TITLE
[RS-414 - PART 2] - Rs 452 mock productdisplayitem

### PR DESCRIPTION
**Pull request by aChanEP**
Description:

This is a pr against: #340 

It mocks all http request the `productdisplayitem.main.jsx` component makes.  This PR also adds dummy functions to pass into the component to highlight more of its functionality.

This component also moves the authentication mock response to a common place as it is used for most components that make http requests.

Linting:

- [x] No linting errors

Tests:

- [ ] E2E tests (npm test run with `e2e`)
- [ ] Manual tests

Documentation:

- [ ] Requires documentation updates

### CODE REVIEW

File: components/src/CommonMockHttpResponses/login_response.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+	"access_token": "fa702e32-c9b9-4b04-bdd4-8152f1fc1439",
+	"token_type": "bearer",
+	"expires_in": 604799,
+	"scope": "VESTRI_B2C",
+	"role": "PUBLIC",
+	"roles": ["PUBLIC"]
+}
```
Comment (shaunmaharaj) [inline components/src/CommonMockHttpResponses/login_response.json]: Please add new lines to each of your files

**Comment by shaunmaharaj** [comment]
No further comments. Thanks for this @aChanEP!!

File: components/src/CommonMockHttpResponses/login_response.json (hunk 1 above)
Comment (shaunmaharaj) [reply components/src/CommonMockHttpResponses/login_response.json]: @aChanEP This one is still missing a new line 😃

Comment (aChanEP) [reply components/src/CommonMockHttpResponses/login_response.json]: kk updating

**Pull request merged**